src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicPreferDictionaryTryGetValueFixer.vb

```diff
@@ -16,60 +16,123 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
+                        Else
+                            dictionaryAccessors.Add(DirectCast(node, InvocationExpressionSyntax))
+                        End If
+                    Case GetType(AssignmentStatementSyntax)
+                        If addStatementNode IsNot Nothing Then
+                            Return
+                        End If
 
-                                      Dim ifStatement As SyntaxNode = containsKeyAccess.AncestorsAndSelf().OfType(Of MultiLineIfBlockSyntax).FirstOrDefault()
-                                      If ifStatement Is Nothing
-                                          ifStatement = containsKeyAccess.AncestorsAndSelf().OfType(Of SingleLineIfStatementSyntax).FirstOrDefault()
-                                      End If
+                        Dim assignment = DirectCast(node, AssignmentStatementSyntax)
+                        changedValueNode = assignment.Right
+                        addStatementNode = assignment
+                    Case Else
+                        Return
+                End Select
+            Next
 
-                                      If ifStatement Is Nothing
-                                          ' For ternary expressions, we need to add the value assignment before the parent of the expression, since the ternary expression is not an alone-standing expression. 
-                                          ifStatement = containsKeyAccess.AncestorsAndSelf().OfType(Of TernaryConditionalExpressionSyntax).FirstOrDefault()?.Parent
-                                      End If
+            Dim targetAmount = diagnostic.AdditionalLocations.Count
+            If addStatementNode IsNot Nothing Then
+                targetAmount -= 1
+            End If
+
+            If targetAmount <> dictionaryAccessors.Count Then
+                Return
+            End If
+
+            Dim semanticModel = Await context.Document.GetSemanticModelAsync(context.CancellationToken).
+                    ConfigureAwait(False)
+            Dim dictionaryValueType = GetDictionaryValueType(semanticModel, dictionaryAccessors(0).Expression)
+            Dim replaceFunction =
+                    Async Function(ct As CancellationToken) As Task(Of Document)
+                        Dim editor = Await DocumentEditor.CreateAsync(document, ct).ConfigureAwait(False)
+                        Dim generator = editor.Generator
+
+                        Dim tryGetValueAccess = generator.MemberAccessExpression(containsKeyAccess.Expression,
+                                                                                 TryGetValue)
+                        Dim keyArgument = containsKeyInvocation.ArgumentList.Arguments.FirstOrDefault()
+                        Dim valueAssignment = generator.LocalDeclarationStatement(dictionaryValueType, Value).
+                                WithLeadingTrivia(SyntaxFactory.EndOfLine(Environment.NewLine)).
```

Comment:
Try to use `ElasticCarriageReturnLineFeed`.
I don't think `Environment.NewLine` should be used.

Reply:
For some reason this adds a line break but also an empty newline after `Dim value As Integer`.
Is there a better way to do it or is that how visual basic wants to be formatted?